nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2589 @@
+            image_terms = {}
+        if max_images is not None and max_images < 0:
+            max_images = None
+        self.collection_terms_ = collection_terms
+        self.collection_filter_ = collection_filter
+        self.image_terms_ = image_terms
+        self.image_filter_ = image_filter
+        self.visited_images_, self.visited_collections_ = set(), set()
+        self.max_images_ = max_images
+        self.max_consecutive_fails_ = max_consecutive_fails
+        self.max_fails_in_collection_ = max_fails_in_collection
+        self.consecutive_fails_ = 0
+        self.fails_in_collection_ = 0
+        self.batch_size_ = batch_size
+
+        if download_manager is None:
+            download_manager = BaseDownloadManager(
+                neurovault_data_dir=self.neurovault_dir_,
+                max_images=self.max_images_)
+        self.download_manager_ = download_manager
+        self.download_manager_.neurovault_data_dir_ = self.neurovault_dir_
+        self.wanted_image_ids_ = wanted_image_ids
+        self.wanted_collection_ids_ = wanted_collection_ids
+        self._prepare()
+
+    def _prepare(self):
+
+        if (self.wanted_collection_ids_ is not None or
+            self.wanted_image_ids_ is not None):
+
+            self.max_images_ = None
+            self.download_manager_.max_images_ = None
+            self.scroll_mode_ = 'explicit'
+            self.image_terms_ = {}
+            self.image_filter_ = _empty_filter
+            self.collection_terms_ = {}
+            self.collection_filter_ = _empty_filter
+            self.local_collection_filter_ = _empty_filter
+            self.local_image_filter_ = ResultFilter(
+                {'id': IsIn(*self.wanted_image_ids_)}).OR(
+                    ResultFilter(
+                        collection_id=IsIn(*self.wanted_collection_ids_)))
+
+            return
+
+        (self.collection_terms_,
+         self.collection_filter_) = _move_unknown_terms_to_local_filter(
+             self.collection_terms_, self.collection_filter_,
+             _COL_FILTERS_AVAILABLE_ON_SERVER)
+
+        (self.image_terms_,
+         self.image_filter_) = _move_unknown_terms_to_local_filter(
+             self.image_terms_, self.image_filter_,
+             _IM_FILTERS_AVAILABLE_ON_SERVER)
+
+        self.local_collection_filter_ = ResultFilter(
+            **self.collection_terms_).AND(self.collection_filter_)
+        self.local_image_filter_ = ResultFilter(**self.image_terms_).AND(
+            self.image_filter_)
+
+        self.scroll_mode_ = 'filtered'
+
+    def _failed_download(self):
+        self.consecutive_fails_ += 1
+        if self.consecutive_fails_ == self.max_consecutive_fails_:
+            _logger.error('Too many failed downloads: {}; '
+                          'stop scrolling remote data.'.format(
+                              self.consecutive_fails_))
+            raise RuntimeError(
+                '{0} consecutive bad downloads'.format(
+
```

Comment:
What does the "impl" stand for in the name of this function? I don't understand it.